src/pudl/transform/eia.py

```diff
@@ -1173,12 +1173,13 @@ def transform(
     # Remove fields that came from input data but aren't in the
     # corresponding SQLite tables. The data may still exist but has been
     # moved elsewhere.
-    eia_transformed_dfs["generation_fuel_eia923"] = eia_transformed_dfs[
-        "generation_fuel_eia923"
-    ].drop(columns=["utility_name_eia"])
-    eia_transformed_dfs["generation_fuel_nuclear_eia923"] = eia_transformed_dfs[
-        "generation_fuel_nuclear_eia923"
-    ].drop(columns=["utility_name_eia"])
+    for cat in eia_transformed_dfs:
+        resource = (
+            pudl.metadata.classes.Package.from_resource_ids().
+            get_resource(cat)
+        )
+        eia_transformed_dfs[cat] = resource.enforce_schema(
+            eia_transformed_dfs[cat])
```

Comment:
Is there any reason not to wait to do this until the very end (after the balancing authority fix below)?

Is there any reason not to call `enforce_schema()` on the entity tables too?

Reply:
Yes, I had already moved the code to the very end of the function.

And I added code to do the same thing for the entity tables. That doesn't seem to cause any distress.

Reply:
Wow that's great! I'm slightly surprised. Do you want to try running `tox -e nuke` overnight and see if stays happy?

Reply:
Will do

Reply:
Seems not successful:
nuke: exit -9 (641.56 seconds) /home/knordback/Kurt/pudl> bash -c 'coverage run --append src/pudl/cli.py --logfile tox-nuke.log --clobber src/pudl/package_data/settings/etl_full.yml' pid=793582
.pkg: _exit> python /home/knordback/.conda/envs/pudl-dev/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta
  nuke: FAIL code -9 (7080.40=setup[98.70]+cmd[0.12,0.54,0.48,0.39,0.41,0.42,0.39,0.53,0.24,6.33,6.21,1.24,0.03,1.72,1.23,161.77,33.89,25.29,3980.21,0.01,2118.71,641.56] seconds)
  evaluation failed :( (7081.04 seconds)

Reply:
Are there logs from the attempt?  Did the earlier tests that get run pass successfully? Have the tests been passing locally in general?

Reply:
The same thing happens on dev, at what looks to be the exact same point in the run. I assume you can run this successfully? If so it must be something in my setup. Looking back at all the messages there are some warnings and a couple of things that say CRITICAL, but insofar as I can tell they're not related.

Reply:
The CI is passing on this branch (all checks passed below) and the nightly builds are passing (which do something very similar to tox -e nuke) so my guess is it's not a problem with the source code. Maybe the environment?  Have you rebuilt your conda environment lately? Did merging in `dev` from upstream make any difference?

Are there any tests you're able to run locally?  E.g. can you run any of:
```sh
pytest test/unit
pytest test/integration/etl_test.py::test_pudl_engine
tox -re unit
tox -re integration
```

Are you not getting copious logging output when you run the tests, indicating what's going on?

Reply:
All of those tests run successfully. So does `tox` with no arguments (which I thought was the commit criterion). So it seems to be something specific to the "nuke".

I am indeed getting copious output, but it's not (to my eye) indicating what the problem is. As mentioned above, I get some warnings but nothing obviously connected to this.

I'll be away next week. I'll will look into this more when I get back.

Reply:
Hi Zane. I'm back, for a little while, and have some time to work on this again. I can continue un-dropping fields, but I'm not sure how worried to be about not being able to run `tox -e nuke` successfully. As noted, the other tests run fine for me. Output is attached in case you want to take a look.
[tox-out.txt](https://github.com/catalyst-cooperative/pudl/files/10993919/tox-out.txt)

Reply:
Have you tried running the full ETL with the `pudl_etl` command line tool? The place that it's failing in your logs is the most memory intensive part of the ETL so my guess is it's running out of memory and crashing? I thought you'd been able to do the full ETL previously?  How much memory + swap do you have? I think it may take up to 25GB as it is now.

`tox` with no arguments is what gets run in CI on GitHub. It only processes 1 year of data. For changes like the one you're making that will affect all years of data we try and run the full ETL locally before it gets merged into `dev`.

`tox -e nuke` is a blunt instrument that not only runs the full ETL, but also all the unit + integration tests on all the data + data validation tests. You probably want to be trying to run the full ETL, but only for the EIA data, to check whether everything is working. The `devtools/eia-etl-debug.ipynb` notebook is the easiest / most efficient way to do that right now, since you don't have to run the extract step over and over again. It'll also avoid this (FERC 1) related memory intensive step.

Reply:
I have 16GB ram + 15GB swap. I thought I could run the full ETL but apparently not that either. I was able to run `devtools/eia-etl-debug.ipynb` (first time using Jupyter -- it's kinda slick).

Reply:
Jupyter is great for working with data. Lots more human-friendly tools for seeing what's going on in there! Definitely worth getting familiar with.